### DISCUSSION TITLE
How to change the search query parameters regardless of current route?

### DISCUSSION THREAD

**Discussion by nuschk** [TanStack/router#2620] [Q&A] [answered]
First, my **use case**: there is a list of items on `/items`  and there is a detail view on `/items/$id`. The list and detail views are shown on the same screen (e.g. a messenger screen).

Now, the list is sortable either by "asc" or "desc". That sorting state should go into the URL, into `?order`. When the user now navigates to a certain item (or in between items), the order parameter is preserved:

```tsx
navigate({ to: "items/$id", params: { id: 12 }, search: true })
```

All good so far. 

Now my question: How can the ordering dropdown component change the URL's `order` search parameter? I would want to modify the search query **regardless of which route we're currently on**:

```tsx
navigate({ search: prev => ({ ...prev, ordering: "asc" }) })
```

But, obviously, that doesn't type-check (although it works at runtime). 

Is there some way to get this to work? Or some pattern I'm missing?

Tanner himself made the following comment 1.5years back, so I don't think that's still valid? https://github.com/TanStack/router/discussions/507#discussioncomment-4789394

**Comment by BrendanC23** [comment, marked as answer]
Does this work?

```ts
navigate({ to: ".", search: prev => ({ ...prev, ordering: "asc" }) });
```

**Reply by nuschk** [reply]
Huh, didn't know that one. It does solve it in that it types `search` to `any`. At least I don't have to `// @ts-ignore`. I was hoping I could specify a range of routes somehow, but that might get too complicated anyway.

Thank you!
